framework/log/Target.php

```diff
@@ -161,6 +162,54 @@ public function collect($messages, $final)
         }
     }
 
+    /**
+     * Flattens a multidimensional array into a one-dimensional array.
+     *
+     * This method recursively traverses the input array and concatenates the keys
+     * to form a new key in the resulting array.
+     *
+     * Example:
+     *
+     * ```php
+     * $array = [
+     *      'A' => [1, 2],
+     *      'B' => [
+     *          'C' => 1,
+     *          'D' => 2,
+     *      ],
+     *      'E' => 1,
+     *  ];
+     * $result = \yii\log\Target::flatten($array);
+     * // result will be:
+     * // [
+     * //     'A.0' => 1
+     * //     'A.1' => 2
+     * //     'B.C' => 1
+     * //     'B.D' => 2
+     * //     'E' => 1
+     * // ]
+     * ```
+     *
+     * @param array $array the input array to be flattened.
+     * @param string $prefix the prefix to be added to each key in the resulting array.
+     *
+     * @return array the flattened array.
+     */
+    private static function flatten($array, $prefix = ''): array
```

Comment:
I'm not sure this method should be here.

Reply:
Write it via Google Translate or something alike. We'll fix the language.

Reply:
Looks like a good candidate for array helper.

Reply:
Can I leave as is for now and then handle it in a separate PR? (dedicated to new `ArrayHelper::flatten()` method)

Reply:
Yes.